src/main/java/io/vertx/core/file/impl/AsyncFileImpl.java

```diff
@@ -418,11 +438,16 @@ public void failed(Throwable exc, Object attachment) {
     });
   }
 
-  private void doRead(Buffer writeBuff, int offset, ByteBuffer buff, long position, Handler<AsyncResult<Buffer>> handler) {
+  private void doRead(Buffer writeBuff, int offset, ByteBuffer buff, long position, long readLength, 
```

Comment:
it would be better here to have the `doRead` method not care about the total length but only care about how much data it should read, i.e transform `readLength` into a `readSize` that is always > 0 and equals to `min(readBufferSize, readLength)`

Reply:
@vietj good hint, the code gets simpler in this way.

Reply:
yes it seesm nicer this way